warehouse/query-core/src/test/java/datawave/query/util/sortedset/BufferedFileBackedSortedSetTest.java

```diff
@@ -0,0 +1,233 @@
+            set.remove(data[i]);
+            expectedSize--;
+            assertEquals(expectedSize, set.size());
+        }
+        for (int i = 0; i < (data.length / 2); i++) {
+            set.remove(data[i]);
+            expectedSize--;
+            assertEquals(expectedSize, set.size());
+        }
+        assertEquals(0, set.size());
+        for (int i = 0; i < data.length; i++) {
+            set.add(data[i]);
+            expectedSize++;
+            assertEquals(expectedSize, set.size());
+        }
+    }
+    
+    @Test
+    public void testIsEmpty() {
+        assertFalse(set.isEmpty());
+        for (int i = (data.length / 2); i < data.length; i++) {
+            set.remove(data[i]);
+            assertFalse(set.isEmpty());
+        }
+        for (int i = 1; i < (data.length / 2); i++) {
+            set.remove(data[i]);
+            assertFalse(set.isEmpty());
+        }
+        set.remove(data[0]);
+        assertTrue(set.isEmpty());
+        for (int i = 0; i < data.length; i++) {
+            set.add(data[i]);
+            assertFalse(set.isEmpty());
+        }
+    }
+    
+    @Test
+    public void testClear() {
+        set.clear();
+        assertTrue(set.isEmpty());
+        for (int i = 0; i < data.length; i++) {
+            set.add(data[i]);
+            assertFalse(set.isEmpty());
+        }
+    }
+    
+    @Test
+    public void testContainsObject() {
+        for (int i = (data.length / 2); i < data.length; i++) {
+            set.remove(data[i]);
+        }
+        for (int i = 1; i < (data.length / 2); i++) {
+            assertTrue(set.contains(data[i]));
+        }
+        for (int i = (data.length / 2); i < data.length; i++) {
+            assertFalse(set.contains(data[i]));
+        }
+    }
+    
+    @Test
+    public void testIterator() {
+        int index = 0;
+        for (Iterator<byte[]> it = set.iterator(); it.hasNext();) {
+            byte[] value = it.next();
+            byte[] expected = data[sortedOrder[index++]];
+            assertTrue(Arrays.equals(expected, value));
+        }
+        set.clear();
+        for (byte[] value : set) {
+            fail();
+        }
+    }
+    
+    @Test
+    public void testIteratortRemove() {
+        int size = set.size();
+        int failCount = 0;
+        for (Iterator<byte[]> it = set.iterator(); it.hasNext();) {
+            byte[] value = it.next();
+            assertTrue(set.contains(value));
+            try {
+                it.remove();
+                assertFalse(set.contains(value));
+                size--;
+                assertEquals(size, set.size());
+            } catch (Exception e) {
+                // expected that some of the underlying FileSortedSets are persisted and hence the remove will fail
+                failCount++;
+                assertTrue(set.contains(value));
+                assertEquals(size, set.size());
+            }
+        }
+        assertTrue(failCount > 0);
```

Comment:
There is nothing that verifies we successfully removed any. The failCount could equal the initial `set.size()`. Do we want to ensure at least one remove succeeded?

Reply:
that is reasonable.....